cmake/module/AddBoostIfNeeded.cmake

```diff
@@ -48,5 +49,53 @@ function(add_boost_if_needed)
     endif()
   endif()
 
-  mark_as_advanced(Boost_INCLUDE_DIR)
+  if(WITH_EXTERNAL_SIGNER)
+    include(CheckCXXSourceCompiles)
+    set(CMAKE_REQUIRED_INCLUDES ${Boost_INCLUDE_DIR})
+    set(CMAKE_REQUIRED_FLAGS "${working_compiler_werror_flag}")
+    # Boost 1.78 requires the following workaround.
+    # See: https://github.com/boostorg/process/issues/235
+    string(APPEND CMAKE_REQUIRED_FLAGS " -Wno-error=narrowing")
```

Comment:
By default, CMake [treats](https://cmake.org/cmake/help/latest/prop_tgt/NO_SYSTEM_FROM_IMPORTED.html)
> include directories from the interfaces of consumed [Imported Targets](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#imported-targets) as system directories.

That means that we have `--enable-suppress-external-warnings` behaviour for granted.

If we won't bother to mirror the opposite behaviour, these 4 lines might be dropped.